smalltalksrc/VMMaker/DruidJIT.class.st

```diff
@@ -5124,20 +5120,20 @@ DruidJIT >> gen_PrimitiveNewWithArg [
 	jump12 jmpTarget: currentBlock.
 	self CmpCq: 0 R: Extra2Reg.
 	jump12 := self JumpNonZero: 0.
-	self MoveR: SendNumArgsReg R: ClassReg.
+	self MoveR: Extra1Reg R: ClassReg.
 	currentBlock := self Label.
 	jump11 jmpTarget: currentBlock.
 	self MoveMw: 0 r: SPReg R: Extra3Reg.
-	self MoveM32: 4 r: Extra3Reg R: Extra2Reg.
-	self AndCq: 16rFFFFFFFF R: Extra2Reg.
-	self AndCq: 16r3FFFFF R: Extra2Reg.
-	self CmpCq: 0 R: Extra2Reg.
+	self MoveM32: 4 r: Extra3Reg R: Extra1Reg.
+	self AndCq: 16rFFFFFFFF R: Extra1Reg.
+	self AndCq: 16r3FFFFF R: Extra1Reg.
+	self CmpCq: 0 R: Extra1Reg.
```

Comment:
Here we have two ANDs but the first one includes the second one. So it can be removed.
Also we could collapse logical and bit AND/OR instructions with comparisons!

Reply:
https://github.com/Alamvic/druid/issues/105 and https://github.com/Alamvic/druid/issues/106